src/modulus_convert.rs

```diff
@@ -0,0 +1,548 @@
+pub mod modulus_conversion {
+
+    use rand::Rng;
+    use rand::RngCore;
+
+    pub struct ReplicatedBinarySecretSharing {
+        pub sh1: u8,
+        pub sh2: u8,
+    }
+
+    pub struct ReplicatedSecretSharing {
+        pub sh1: u64,
+        pub sh2: u64,
+    }
+
+    pub fn gen_r_binary_sharings<R: RngCore>(
+        mut rng_1: R,
+        mut rng_2: R,
```

Comment:
Rather than passing in just the RNGs that you need, would it make more sense for there to be ambient information that had `rng_left` (for $\mathsf{P}_{i-1}$ and `rng_right` (for $\mathsf{P}_{i+1}$)? 

Edit: Gah, no math support on comments... :(